build.gradle.kts

```diff
@@ -73,9 +73,13 @@ java {
 
 tasks {
     val generateVersionProperties = register("generateVersionProperties") {
+        val projectVersion = version
+        val propertiesFile = File(sourceSets.main.get().output.resourcesDir, "version.properties")
+        inputs.property("projectVersion", projectVersion)
+        outputs.file(propertiesFile)
```

Comment:
I couldn’t find a reason why modify the project sources was needed, so I thought I can propose to modify the `build` directory directly. 
I verified the file generates under the same location:

<img width="385" alt="image" src="https://user-images.githubusercontent.com/36954793/159121989-fb119de4-415c-4843-b666-fe12a1506b73.png">

and it can be found in published jar:
<img width="454" alt="image" src="https://user-images.githubusercontent.com/36954793/159122127-b7ac8b57-8110-43ea-9181-22254e9cc834.png">

If my assumptions are invalid, previous behaviour would be:
```suggestion
        val projectVersion = version
        val propertiesFile = File(sourceSets.main.get().resources.sourceDirectories.asPath, "version.properties")
        inputs.property("projectVersion", projectVersion)
        outputs.file(propertiesFile)
```